algorithms/src/std_algorithms/impl/Kokkos_RandomAccessIterator.hpp

```diff
@@ -111,77 +133,91 @@ class RandomAccessIterator< ::Kokkos::View<DataType, Args...> > {
 
   KOKKOS_FUNCTION
   reference operator[](difference_type n) const {
-    return m_view(m_current_index + n);
+    if constexpr (is_always_contiguous)
+      return *(m_data + n);
+    else
+      return *(m_data + n * m_stride);
   }
 
   KOKKOS_FUNCTION
   iterator_type& operator+=(difference_type n) {
-    m_current_index += n;
+    if constexpr (is_always_contiguous)
+      m_data += n;
+    else
+      m_data += n * m_stride;
     return *this;
   }
 
   KOKKOS_FUNCTION
   iterator_type& operator-=(difference_type n) {
-    m_current_index -= n;
+    if constexpr (is_always_contiguous)
+      m_data -= n;
+    else
+      m_data -= n * m_stride;
     return *this;
   }
 
   KOKKOS_FUNCTION
   iterator_type operator+(difference_type n) const {
-    return iterator_type(m_view, m_current_index + n);
+    auto it = *this;
+    it += n;
+    return it;
   }
 
   KOKKOS_FUNCTION
   iterator_type operator-(difference_type n) const {
-    return iterator_type(m_view, m_current_index - n);
+    auto it = *this;
+    it -= n;
+    return it;
   }
 
   KOKKOS_FUNCTION
   difference_type operator-(iterator_type it) const {
-    return m_current_index - it.m_current_index;
+    if constexpr (is_always_contiguous)
+      return m_data - it.m_data;
+    else
+      return (m_data - it.m_data) / m_stride;
   }
 
   KOKKOS_FUNCTION
   bool operator==(iterator_type other) const {
-    return m_current_index == other.m_current_index &&
-           m_view.data() == other.m_view.data();
+    return m_data == other.m_data && m_stride == other.m_stride;
   }
 
   KOKKOS_FUNCTION
   bool operator!=(iterator_type other) const {
-    return m_current_index != other.m_current_index ||
-           m_view.data() != other.m_view.data();
+    return m_data != other.m_data || m_stride != other.m_stride;
```

Comment:
I haven't thought too much about it and I see that it was the previous behavior but it disturbs me that we can have two iterators that do not compare equal nor compare "not equal" (that is both `it1 == it2` and `it1 != it2` can be `false` at the same time)

Reply:
I'm not sure what you are talking about.
`!(m_data == other.m_data && m_stride == other.m_stride)` is identical to `m_data != other.m_data || m_stride != other.m_stride`. What am I missing?

Reply:
Misread. Sorry about the noise.